pkg/internal/cluster/kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"bytes"
+
+	"sigs.k8s.io/kind/pkg/cluster/nodeutils"
+	"sigs.k8s.io/kind/pkg/errors"
+	"sigs.k8s.io/kind/pkg/internal/cluster/context"
+
+	"sigs.k8s.io/kind/pkg/internal/util/kubeconfig"
+)
+
+// Export exports the kubeconfig given the cluster context and a path to write it to
+// This will always be an external kubeconfig
+func Export(ctx *context.Context, explicitPath string) error {
+	cfg, err := get(ctx, true)
+	if err != nil {
+		return err
+	}
+	return kubeconfig.WriteMerged(cfg, explicitPath)
+}
+
+// Get returns the kubeconfig for the cluster
+// external controls if the internal IP address is used or the host endpoint
+func Get(ctx *context.Context, external bool) (string, error) {
+	cfg, err := get(ctx, external)
+	if err != nil {
+		return "", err
+	}
+	b, err := kubeconfig.Encode(cfg)
+	if err != nil {
+		return "", err
+	}
+	return string(b), err
+}
+
+func Context(kindClusterName string) string {
+	return kubeconfig.KINDClusterKey(kindClusterName)
```

Comment:
is this passing the linter with all CAPS?

Reply:
yes, acronyms are same case

Reply:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kind/1029/pull-kind-verify/1189674919814762498